examples/nanocoap_server/coap_handler.c

```diff
@@ -6,21 +6,52 @@
  * directory for more details.
  */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
 #include "nanocoap.h"
 
+/* internal value that can be read/written via CoAP */
+static uint8_t internal_value = 0;
+
 static ssize_t _riot_board_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len)
 {
     return coap_reply_simple(pkt, buf, len,
             COAP_FORMAT_TEXT, (uint8_t*)RIOT_BOARD, strlen(RIOT_BOARD));
 }
 
+static ssize_t _riot_value_handler(coap_pkt_t *pkt, uint8_t *buf, size_t len)
+{
+    ssize_t p = 0;
+    char rsp[16];
+
+    /* read coap method type in packet */
+    unsigned method_flag = coap_method2flag(coap_get_code_detail(pkt));
+
+    switch(method_flag) {
+    case COAP_GET:
+        /* write the response buffer with the internal value */
+        p += sprintf(rsp, "value: %d", internal_value);
```

Comment:
why be so _chatty_ here, if I call `GET /riot/value` I would expect to get the plain value. Otherwise I would have to parse the answer again to get the actual value. Hence, make it `p += fmt_u32_dec(rsp, internal_value);`.